editor/src/messages/tool/common_functionality/shape_editor.rs

```diff
@@ -461,6 +463,16 @@ impl ShapeState {
 			} else {
 				// Merge the layers
 				merge_layers(document, layer1, layer2, responses);
+
+				let resolve_id = |layer: LayerNodeIdentifier, point_id: PointId, index: usize| {
+					let mut hasher = DefaultHasher::new();
+					(index, 0_usize, layer).hash(&mut hasher);
+					point_id.generate_from_hash(hasher.finish())
+				};
+
+				let start_point = resolve_id(layer1, start_point, 0);
+				let end_point = resolve_id(layer2, end_point, 1);
```

Comment:
Only ids in `layer2` will change (since the `layer1` will probably have no conflicts).

Furthermore it is possible that there is no conflict, in which case the original `PointId` must be used (like before this PR). You therefore need to check if there was some conflict by checking all of the point ids in the first path.

I don't care if this doesn't work for tables with several vector data, since I find them annoying.